edge-client/subdevice_fota.cpp

```diff
@@ -0,0 +1,292 @@
+        FOTA_TRACE_ERROR("Unable to allocate FOTA ctx.");
+        return FOTA_STATUS_OUT_OF_MEMORY;
+
+    }
+    fota_ctx->fw_info = (manifest_firmware_info_t*) calloc(1, sizeof(manifest_firmware_info_t));
+    if (!fota_ctx->fw_info) {
+        FOTA_TRACE_ERROR("Unable to allocate FW info.");
+        return FOTA_STATUS_OUT_OF_MEMORY;
+    }
+    endpoint = (char*) calloc(ENDPOINT_SIZE,1*sizeof(char));
+    if(!endpoint) {
+        FOTA_TRACE_ERROR("Unable to allocate memory for endpoint");
+        return FOTA_STATUS_OUT_OF_MEMORY;      
+    }
+    return 0;
+}
+
+int get_component_name(char* c_name) {
+    if(fota_ctx) {
+        memcpy(c_name, fota_ctx->fw_info->component_name, FOTA_COMPONENT_MAX_NAME_SIZE);
+        return 0;
+    }
+    else
+        return 0;
+}
+
+int update_result_resource(char* device_id, uint8_t val) {
+
+    return edgeclient_set_resource_value(device_id,
+                                MANIFEST_OBJECT,
+                                MANIFEST_INSTANCE,
+                                MANIFEST_RESOURCE_RESULT,
+                                "",
+                                &val,
+                                sizeof(val),
+                                LWM2M_INTEGER,
+                                1,
+                                NULL);
+
+}
+
+int update_state_resource(char* device_id, uint8_t val) {
+    return edgeclient_set_resource_value(device_id,
+                                MANIFEST_OBJECT,
+                                MANIFEST_INSTANCE,
+                                MANIFEST_RESOURCE_STATE,
+                                "",
+                                &val,
+                                sizeof(val),
+                                LWM2M_INTEGER,
+                                1,
+                                NULL);
+}
+
+void get_endpoint(char* endpoint, const char* uri) {
+    char *uri_path = strdup(uri); // URI : d/device_id/10252/0/1
+    char *left_string = NULL;
+    char *manifest_res = NULL;
+    strtok_r(uri_path, "/", &left_string); // left_string : device_id/10252/0/1
+    char *device_id = strtok_r(left_string, "/", &manifest_res);
+    memcpy(endpoint, device_id, strlen(device_id));
+    if(uri_path) {
+        free(uri_path);
+        uri_path = NULL;
+    }
+}
+
+void get_vendor_id(uint8_t* v_id) {
+    if(fota_ctx) {
+        memcpy(v_id, fota_ctx->fw_info->vendor_id, FOTA_MANIFEST_VENDOR_ID_SIZE);
+    }
+    else {
+        return;
+    }
+}
+void get_class_id(uint8_t* c_id) {
+    if(fota_ctx) {
+        memcpy(c_id, fota_ctx->fw_info->class_id, FOTA_MANIFEST_CLASS_ID_SIZE);
+    }
+    else {
+        return;
+    }
+}
+
+void get_uri(char* c_url) {
+    if(fota_ctx) {
+        memcpy(c_url, fota_ctx->fw_info->uri, FOTA_MANIFEST_URI_SIZE);
+    }
+    else {
+        return;
+    }
+}
+
+
+unsigned int get_component_id() {
+    if(fota_ctx) {
+        return fota_ctx->comp_id;
+    }
+    else {
+        return 0;
+    }
```

Comment:
If the condition is fulfilled then the execution will only reach here in else case, so it could be reduced to just 
```suggestion
    return 0;
```